src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -1328,6 +1327,45 @@ private void CheckConfigurationPaths(RavenConfiguration parentConfiguration, Rav
                                                     $"There is an intersection with database {currentDatabaseName} Temp file located in {currentConfiguration.Indexing.TempPath.FullPath}.");
             }
         }
+
+        private sealed class DatabaseWakeupTimer : IDisposable
+        {
+            private IdleDatabaseActivity _activity;
+
+            private readonly string _databaseName;
+            private readonly Action<string, IdleDatabaseActivity> _callback;
+            private readonly Timer _timer;
+
+            public DatabaseWakeupTimer([NotNull] string databaseName, [NotNull] IdleDatabaseActivity activity, [NotNull] Action<string, IdleDatabaseActivity> callback)
+            {
+                _databaseName = databaseName ?? throw new ArgumentNullException(nameof(databaseName));
+                _activity = activity ?? throw new ArgumentNullException(nameof(activity));
+                _callback = callback ?? throw new ArgumentNullException(nameof(callback));
+
+                _timer = new Timer(TimerCallback, null, _activity.DueTime, Timeout.Infinite);
```

Comment:
you will always create a timer here, right? if so, I think we should do that as lazy.

Reply:
the lazy should be at the `ConcurrentDictionary` level.
https://github.com/ravendb/ravendb/pull/20284/files#diff-80543cdda7318d3299a289707e9a1b3aff2a40f45000f0141774d207edc748bfR43

Reply:
Why? This looks fine, we pass a func to the .AddOrUpdate so no instance will be created unless you are actually adding to the dict that value

Reply:
If 2 threads that are calling `AddOrUpdate` concurrently we might end up with 2 timers.
`Delegates for these methods are called outside the locks to avoid the problems that can arise from executing unknown code under a lock. Therefore, the code executed by these delegates is not subject to the atomicity of the operation.`
https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2?view=net-9.0&redirectedfrom=MSDN#remarks